plugins/ScoutingNanoAOD_miniIso.cc

```diff
@@ -0,0 +1,186 @@
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+
+#include "DataFormats/Math/interface/libminifloat.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Common/interface/View.h"
+
+// Root include files
+#include "TLorentzVector.h"
+#include "TFile.h"
+#include "TTree.h"
+#include "TMath.h"
+#include "TH2F.h"
+#include "TCanvas.h"
+#include "TVectorD.h"
+#include "TMatrixD.h"
+#include "TMatrixDSym.h"
+#include "TMatrixDSymEigen.h"
+
+// User include files
+
+#include "fastjet/ClusterSequence.hh"
+#include "fastjet/ClusterSequenceArea.hh"
+#include "fastjet/contrib/Nsubjettiness.hh"
+#include "fastjet/contrib/XConePlugin.hh"
+#include "fastjet/contrib/SoftDrop.hh"
+#include "fastjet/tools/Pruner.hh"
+#include "fastjet/tools/Filter.hh"
+#include "fastjet/contrib/RecursiveSoftDrop.hh"
+#include "fastjet/contrib/EnergyCorrelator.hh"
+#include "fastjet/JadePlugin.hh"
+#include "fastjet/contrib/SoftKiller.hh"
+
+#include "PhysicsTools/CandUtils/interface/EventShapeVariables.h"
+#include "PhysicsTools/CandUtils/interface/Thrust.h"
+
+#include "ScoutingNanoAOD_miniIso.h"
+
+using namespace std;
+using namespace fastjet;
+using namespace fastjet::contrib;
+
+
+/*
+  Defines a function to compute MiniIsolation given a 4-vector and a collection
+  of packed PF candidates.
+
+  Mini-Isolation reference: https://hypernews.cern.ch/HyperNews/CMS/get/susy/1991.html
+
+*/
+
+//Implementation of MiniIsolation for Scouting (following https://github.com/cms-sw/cmssw/blob/0cbc4b00b1664496105480e9f561a29ba7c9a68b/PhysicsTools/PatUtils/interface/MiniIsolation.h)
+float miniIsoDrScout(const reco::Candidate::PolarLorentzVector &p4, float mindr, float maxdr, float kt_scale) {
+    return std::max(mindr, std::min(maxdr, float(kt_scale / p4.pt())));
+  };
+
+pat::PFIsolation getMiniPFIsolationScout(const vector<ScoutingParticle> pfcands,
+                                const reco::Candidate::PolarLorentzVector &p4,
+                                float mindr = 0.05,
+                                float maxdr = 0.2,
+                                float kt_scale = 10.0,
+                                float ptthresh = 0.5,
+                                float deadcone_ch = 0.0001,
+                                float deadcone_ph = 0.01,
+                                float deadcone_nh = 0.01
+                                ) {
+    
+    float chiso = 0, nhiso = 0, phiso = 0 , puiso = 0;
+    float drcut = miniIsoDrScout(p4, mindr, maxdr, kt_scale);
+    for (const auto &pc : pfcands) {
+      float dr2 = deltaR2(p4, pc);
+      if (dr2 > drcut * drcut)
+        continue;
+      float pt = pc.pt();
+      int id = pc.pdgId();
+      if (std::abs(id) == 211) {
+        if (dr2 > deadcone_ch * deadcone_ch) {
+          // if charged hadron and from primary vertex, add to charged hadron isolation
+          chiso += pt;
+        } 
```

Comment:
Where are you checking that the ch is from the PV?

Reply:
yes, that is an old comment from the implementation that we have in miniAOD, but at the moment is not checked. Do you think we can use this variable here: https://github.com/cms-sw/cmssw/blob/f6e0fc429ceb8c78f0f0391bcc915e696065122d/DataFormats/Scouting/interface/ScoutingParticle.h#L22 ?